File: 0328-odd-even-linked-list/0328-odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def oddEvenList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        odd = oddHead = ListNode()
        even = evenHead = ListNode()
        isOdd = True

        while head:
            if isOdd:
                odd.next = head
                odd = head
            else:
                even.next = head
                even = head
            
            head = head.next
            isOdd = not isOdd
        
        even.next = None
        odd.next = evenHead.next
        
        return oddHead.next

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func oddEvenList(head *ListNode) *ListNode {
    left, right := &ListNode{
        Next: head,
    }, &ListNode{
        Next: head,
    }
    curLeft, curRight := left, right
    cnt := 0
    for head != nil {
        cnt += 1
        if cnt%2 == 1 {
            curLeft.Next = head
            curLeft = curLeft.Next
        } else {
            curRight.Next = head
            curRight = curRight.Next
        }

        head = head.Next
    }

    curRight.Next = nil
    curLeft.Next = right.Next
    return left.Next
}